frontend/packages/pipelines-plugin/src/components/pipelines/modals/common/utils.ts

```diff
@@ -112,7 +112,9 @@ export const getPipelineRunData = (
             generateName: `${pipelineName}-`,
           }
         : {
-            name: `${pipelineName}-${getRandomChars()}`,
+            name: latestRun
+              ? `${latestRun.metadata.name}-${getRandomChars()}`
+              : `${pipelineName}-${getRandomChars()}`,
```

Comment:
We should not just append random chars. This will result in longer PipelineRun names each time you rerun it:

my-pr-123456
my-pr-123456-123456
my-pr-123456-123456-123456
etc.

Instead of that, we should create a function similar to `getPipelineName` that returns a PipelineRun prefix based on these rules (my suggestion, feel free to fine tune it):

1. `latestRun.metadata.annotations?.[preferredNameAnnotation]` (as sawn in the getPipelineName) ?? Does this makes sense ??
2. `latestRun.metadata.generateName` (when you create a PipelineRun with `generateName` this saves the origin name like "my-pipeline-"
3. `latestRun.metadata.name.replace(/-[a-z]{5,6}$/, '')`

   Why drop 5 or 6 characters? Because we create PR names with `getRandomChars` with 6 random characters, while Kubernetes create resources with 5 characters when using `generateName`.